### PR TITLE
Update dbtools

### PR DESCRIPTION
**Pull request by strempfer**
implement new function to read baseline devices

### CODE REVIEW

File: polartools/load_data.py [hunk 1]
```diff
@@ -208,3 +216,256 @@ def is_Bluesky_specfile(source, folder=""):
...
+    Returns
+    -------
+    output: list
+
+    """
+
+    db_range = db_query(db, query=query) if query else db
+
+    status = db_range[scan].metadata["stop"]
+    if status and status["exit_status"] == "success":
```
Comment (gfabbris) [inline polartools/load_data.py:447]: what if the scan was aborted? We should be able to read the baseline anyway.

Comment (gfabbris) [reply polartools/load_data.py:447]: Even if `status = None`, the baseline still has the first value.

File: polartools/load_data.py [hunk 2]
```diff
@@ -208,3 +216,256 @@ def is_Bluesky_specfile(source, folder=""):
...
+        baseline = load_databroker(scan, db, "baseline")
```
Comment (gfabbris) [inline polartools/load_data.py:448]: We need to be a bit careful here. It turns out, the way that the `databroker.v1` and `databroker.v2` load the `dataframe` is different (if the v2 is used, this function will not work). So I suggest that you replace line 448 to:
```python
# Enforces use of v1. Will need to update this function if v2 loads faster in the future.
baseline = load_databroker(scan, db, "baseline", use_db_v1=True)
```

File: polartools/load_data.py [hunk 3]
```diff
@@ -208,3 +216,256 @@ def is_Bluesky_specfile(source, folder=""):
...
+            if search_string in key:
+                if isinstance(baseline[key][1], list):
+                    print(f"{key:>50}{baseline[key][1]}{baseline[key][2]}")
+                else:
+                    print(
+                        f"{key:>50}{baseline[key][1]:>25}{baseline[key][2]:>25}"
+                    )
+
+    elif not status:
+        raise ValueError("Scan not existing!")
```
Comment (gfabbris) [inline polartools/load_data.py:464]: I think this is not true, the baseline may exist (with 1 point), but `status = None`

**Review by gfabbris — CHANGES_REQUESTED** [review]
See comments. The main thing is that we may want to show the baseline even if the stop document doesn't exist, or if the scan is not "success". Let me know if you want me to try to address this.

File: polartools/load_data.py (hunk 1 above)
Comment (strempfer) [reply polartools/load_data.py:447]: There are examples for aborted scans (429) and `status = None` scans (430), where no baseline information in available. If one is really desperate and needs this information, one can dig into it directly.

File: polartools/load_data.py (hunk 3 above)
Comment (strempfer) [reply polartools/load_data.py:464]: Changed wording in if none statement to 'possibly' as well. It is not possible to take care of all possibilities directly. If this error message appears direct access of the baseline may still reveal some information.

**Comment by gfabbris** [comment]
I think you don't need to check the exit status. Just check if the baseline stream exists, and print it if it does.

**Comment by strempfer** [comment]
How do you check on the baseline if it does not exist without raising a ValueError?

**Comment by gfabbris** [comment]
It's fine for a value error to pop up if the stream doesn't exist, but it'd be good to have a good message for what the problem is, so I think this would be best done in `load_databroker`:

```python
def load_databroker(scan_id, db, stream="primary", query=None, use_db_v1=True):
    """
    Load data of the first scan with the provided scan_id.

    Currently defaults to databroker.v1 because it is faster. See issue #28.

    For further details, refer to the `databroker` `documentation`_.

    .. _documentation: https://blueskyproject.io/databroker/

    Parameters
    ----------
    scan_id : int
        Scan_id of the scan to be retrieved
    db :
        `databroker` database
    stream : string, optional
        Selects the stream from which data will be loaded.
    query : dict, optional
        Dictionary with search parameters for the database.
    use_db_v1 : bool, optional
        Chooses databroker API version between 'v1' or 'v2', defaults to 'v1'.

    Returns
    -------
    data : pandas.DataFrame
        Table with the data from the primary stream.
    """

    _db = db_query(db, query) if query else db
    if use_db_v1:
        if stream in _db.v1[scan_id].stream_names:
            return _db.v1[scan_id].table(stream_name=stream)
        else:
            raise ValueError(f"The stream {stream} does not exist in scan {scan_id}.")
    else:
        try:
            return getattr(_db.v2[scan_id], stream).read().to_dataframe()
        except AttributeError:
            raise ValueError(f"The stream {stream} does not exist in scan {scan_id}.")
```

**Comment by strempfer** [comment]
great suggestion. This simplifies things a lot.

**Pull request merged**